playbooks/ceph_client.yml

```diff
@@ -1,12 +1,15 @@
 ---
-
 - name: Configure EDPM as client of Ceph
   hosts: "{{ edpm_override_hosts | default('all', true) }}"
   strategy: linear
   gather_facts: "{{ gather_facts | default(false) }}"
   any_errors_fatal: "{{ edpm_any_errors_fatal | default(true) }}"
   max_fail_percentage: "{{ edpm_max_fail_percentage | default(0) }}"
   tasks:
+    - name: Check remote connections
```

Comment:
This won't work when `gather_facts` is true, as it would try gathering facts and fail before calling the task to `wait_for_connection`. That's why it was added as a separate play in [1] with `gather_facts: false`. We regressed that later though.

We either  create a separate play (with `gather_facts: false`) like earlier or change it to false in every playbook (not honor user provided `gather_facts` ansible var at all).

Also, what're we going to do for custom services? Are we not expecting people to use them in bgp deployments? Are we going to document that?

[1] https://github.com/openstack-k8s-operators/edpm-ansible/commit/07a081594bcaff4edc85230663b00f1142351777

Reply:
thanks for the pointer on the gather_facts. I'll update it. 

re custom services, from my pov I think its ok to document it. @slagle whats your take on this?

Reply:
we tried to just pass in the `ANSIBLE_TIMEOUT` env var via the nodeset, but it does not work. Not sure if the `No route to host`  error results in the timeout to be handled as we need it.

```
$ oc logs download-cache-edpm-deployment-r2-compute-nodes-r2-fqvgp                                                                                                                        
Identity added: /runner/artifacts/download-cache-edpm-deployment-r2-compute-nodes-r2/ssh_key_data (cifmw_reproducer_key)                                                                                              
[WARNING]: Invalid characters were found in group names but not replaced, use
-vvvv to see details

PLAY [EDPM Download cache] *****************************************************

TASK [osp.edpm.edpm_download_cache : Validating arguments against arg spec 'main' - The main entry point for the edpm_download_cache role.] ***                                                                       
ok: [r2-compute-ib21ttu9-0]

TASK [osp.edpm.edpm_download_cache : Install role requirements] ****************
included: /usr/share/ansible/collections/ansible_collections/osp/edpm/roles/edpm_download_cache/tasks/install.yml for r2-compute-ib21ttu9-0                                                                           

TASK [Execute bootstrap command] ***********************************************
included: osp.edpm.edpm_bootstrap for r2-compute-ib21ttu9-0

TASK [osp.edpm.edpm_bootstrap : Re-read facts] *********************************
fatal: [r2-compute-ib21ttu9-0]: UNREACHABLE! => {"changed": false, "msg": "Failed to connect to the host via ssh: ssh: connect to host r2-compute-ib21ttu9-0 port 22: No route to host", "unreachable": true}         

NO MORE HOSTS LEFT *************************************************************

NO MORE HOSTS LEFT *************************************************************

PLAY RECAP *********************************************************************
r2-compute-ib21ttu9-0      : ok=3    changed=0    unreachable=1    failed=0    skipped=0    rescued=0    ignored=0   
```

Reply:
@slagle you have other idea for a ansible configuration to test instead of what we tried as mentioned above?

Reply:
` ANSIBLE_SSH_RETRIES` and `ANSIBLE_SSH_TIMEOUT` ( i believe that's the same as `ANSIBLE_TIMEOUT`, but it may be worth trying)

Reply:
tests with setting ANSIBLE_SSH_RETRIES and ANSIBLE_SSH_TIMEOUT were successful. I'll close this pr. Thanks a lot @slagle